File: spec/server/handlers/show_time_stats_spec.rb
# coding: utf-8
require_relative '../spec_helper'

describe SM::Handlers::ShowTimeStats, api: true, vcr: VCR_OPTS do

    let(:show_time) { FactoryBot.create :show_time }
    let(:user) { FactoryBot.create :user }

    it "retrieves show stats" do
        3.times do
            sale = SM::Sale.new(qty: 1, show_time: show_time)
            sale.attendee = SM::Attendee.new(name: 'Tester')
            sale.save!
        end
        show_time.sales.last.redemptions.create!(qty: 1)

        get( "/api/sm/show-time/stats/#{show_time.id}.json", {},
             {'HTTP_AUTHORIZATION' => user.jwt_token})

        expect(last_response).to be_ok
        expect(last_response_json['data']).to(
            include(
                'sales' => 3,
                'redemptions' => 1
            )
        )
    end

end

File: spec/client/models/show-time-stats.spec.js
import ShowTimeStats from 'sm/models/show-time-stats';
import Show from 'sm/models/show';

describe(ShowTimeStats, () => {
    let stats;
    let show;

    beforeEach(() => {
        stats = new ShowTimeStats();
        show = new Show({
            times: [
                { occurs_at: '2018-03-30T15:45:15.000Z' },
            ],
        });
        [stats.showTime] = show.times;
        stats.timeline = [
            [1521773152, 8], [1521773152, 1], [1521773152, 2], [1521773152, 2],
            [1520473821, 1], [1520473821, 4], [1520473821, 1], [1520473821, 3],
            [1520473821, 2], [1520473821, 3], [1520473821, 4], [1520473821, 2],
        ];
    });

    it('#chartData', () => {
        expect(stats.chartData.labels).toEqual([
            '7/3/18', '8/3/18', '9/3/18', '10/3/18', '11/3/18',
            '12/3/18', '13/3/18', '14/3/18', '15/3/18', '16/3/18',
            '17/3/18', '18/3/18', '19/3/18', '20/3/18', '21/3/18', '22/3/18',
        ]);
    });
});
